programs/fileio.c

```diff
@@ -733,20 +734,15 @@ static size_t FIO_createDictBuffer(void** bufferPtr, const char* fileName, FIO_p
 
 #if (PLATFORM_POSIX_VERSION > 0)
 #include <sys/mman.h>
-static void* FIO_mmap(size_t fileSize, int fileHandle)
+static int FIO_munmap(const FIO_Dict_t* dict)
 {
-    return mmap
-    (NULL, (size_t)fileSize, PROT_READ, MAP_PRIVATE, fileHandle, 0);
+    return munmap(dict->dictBuffer, dict->dictBufferSize);
 }
-static int FIO_munmap(void* buffer, size_t bufferSize)
-{
-    return munmap(buffer, bufferSize);
-}
-/* We might want to also do mapping for windows */
-static size_t FIO_createDictBufferMMap(void** bufferPtr, const char* fileName, FIO_prefs_t* const prefs, stat_t* dictFileStat)
+static size_t FIO_createDictBufferMMap(FIO_Dict_t* dict, const char* fileName, FIO_prefs_t* const prefs, stat_t* dictFileStat)
```

Comment:
discussion on naming convention : 

When I read a function name like `PREFIX_createType()`, I expect it to return a `Type*` pointer object. 
That's not the case here.

On a related note, if the object is allocated beforehand (on stack for example), I would consider `PREFIX_initType(Type* v, ...)` for the function name. 

I note that this situation is slightly different : you want to set member `dict->dictBuffer` with the outcome of `mmap` (and I presume `FIO_Dict_t` is larger than this member), _and_ return the underlying file size.

In which case, maybe `FIO_setDictBufferMMap()` ?